### PR TITLE
Hugginface Storage

### PR DESCRIPTION
**Pull request by FrancescoSaverioZuppichini**
This PR adds `HuggingFaceStorage` #24 

Usage

```python
...
storage = HuggingFaceStorage()
model: nn.Module = ...
cfg: Config = ....
storage.put("MODEL_NAME", model.state_dict(), cfg)
state_dict, config = storage.get("MODEL_NAME")
```

### CODE REVIEW

File: glasses/storage/huggingface/__init__.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+        upload_folder(repo_id=repo_id, folder_path=folder_path, path_in_repo=".")
+
+        rmtree(folder_path)
+
+    def get(self, key: str) -> Tuple[StateDict, Config]:
+        cached_dir = f"{self._local_storage.root}/{key}"
+        # create if it doesn't exist
+        Path(cached_dir).mkdir(parents=True, exist_ok=True)
+        clone_from = f"https://huggingface.co/{self.organization}/{key}"
+        # using hf shitty api to instantiate a repo and clone it
```
Comment (AnugunjNaman) [inline glasses/storage/huggingface/__init__.py:43]: 🤣

**Review by AnugunjNaman — APPROVED** [review]
LGTM!

**Comment by AnugunjNaman** [comment]
For the failed test part, it needs personal token can't we skip that test?

**Comment by FrancescoSaverioZuppichini** [comment]
The test is currently pushing to hugging face. We should mock, I've added my personal token to our secrets. Should work :)

**Pull request merged**